### DISCUSSION TITLE
[Bug Report] WALC doesn't recognize microphone

### DISCUSSION THREAD

**Discussion by ubuntuman9** [cstayyab/WALC#63] [Troubleshooting]
mi walc dont reconice mi microphone

**Comment by cstayyab** [comment]
Which version of WALC areyou using? Snap or AppImage? Which Linux Distro are you using? its Version etc? Please provide these details.

**Comment by omaregb** [comment]
Hi, I have the same problem. The mic is not detected. I'm in Manjaro and this is the Snap version. Not sure if it's relevant, but it also didn't work on Whatsapp web on chrome initially, but this was because Chrome automatically detects the wrong device as mic, selecting it manually in Chrome settings fixed it in the browser.

**Comment by cstayyab** [comment]
@omaregb & @ubuntuman9  This might just be a permission issue. Can you please run this in the terminal and tell me about the results:

`sudo snap connect walc:audio-record :audio-record`

The above command should fix the issue.

**Comment by omaregb** [comment]
I get this : 

`error: snap "walc" has no plug named "audio-record"
`

**Comment by cstayyab** [comment]
@omaregb can you post the output of 

`sudo snap connections walc`

**Comment by omaregb** [comment]
sure, this is it. 
![screenhot](https://user-images.githubusercontent.com/34607077/98379218-aa12fe80-200c-11eb-88eb-b9509ae2145b.png)

**Comment by cstayyab** [comment]
Okay, I think I have found the problem. Some of the snap interfaces have been deprecated and updated by others so I will have to replace it with new ones. I hope to fix it over the weekend.

**Comment by cstayyab** [comment]
Electron is not auto-connecting `pulseaudio` interface with the WALC. Please run the following command to get the Microphone working:

```sh
sudo snap connect walc:pulseaudio :pulseaudio
```

CC: @omaregb @ubuntuman9

**Reply by rex2112** [reply]
Hi.

I have an issue with the audio playback (audio and video). I am using Archlinux which as you may know is a "self-service" distro. WALC did work fine under Ubuntu.

Tried the connections command and this is my output:
![Screenshot_2021-02-22_12-39-59](https://user-images.githubusercontent.com/1181639/108754147-30dfde80-750b-11eb-96b8-1b8100de2d37.png)

So I tried the following with no avail:
`sudo snap connect walc:audio-playback :audio-playback`

I am using alsa not pulseaudio. What do you suggest?

Great work by the way. Love the app.

**Comment by omaregb** [comment]
I did this. the command does not return any error messages but after restarting walc microphone is still not recognized
